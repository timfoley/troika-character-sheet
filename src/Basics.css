.basics {
	width: 100%;
	grid-area: basics;
	display: grid;
	grid-gap: 8px;
	grid-template-areas:
		"name"
		"background"
		"special";
}

@media (min-width: 500px) {
	.basics {
		grid-template-columns: 1fr 1fr 1fr;
		grid-template-areas:
			"name background background"
			"special special special ";
	}
}

.basics__name {
	grid-area: name;
}

.basics__background {
	grid-area: background;
}

.basics__input--special {
	font-size: 0.9rem;
}

.basics__special {
	grid-area: special;
}

.basics__input--special {
	height: 180px;
}

label {
	font-weight: bold;
	display: flex;
	flex-direction: column;
}

.basics__input {
	line-height: 1.2rem;
}

.basics input,
textarea {
	padding: 6px;
	border: 1px solid black;
	width: 100%;
}
