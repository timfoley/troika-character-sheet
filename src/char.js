export default {
	name: "Dahddy",
	background: "Beef Steak",
	special: "",
	skill: 5,
	stamina: {
		max: 18,
		remaining: 18
	},
	luck: {
		max: 10,
		remaining: 10
	},
	skills: [
		{
			name: "Hammer Fighting",
			rank: 3
		},
		{
			name: "Fist Fighting",
			rank: 2
		},
		{
			name: "Wrestle",
			rank: 2
		},
		{
			name: "Swim",
			rank: 2
		},
		{
			name: "Climb",
			rank: 2
		},
		{
			name: "Run",
			rank: 2
		},
		{
			name: "Grill",
			rank: 2
		}
	],
	weapons: [
		{
			name: "Steak Knife",
			damage: [2, 2, 2, 2, 4, 8, 10],
			twoHanded: false,
			armorPiercing: false
		},
		{
			name: "Meat Tenderizer",
			damage: [1, 2, 4, 6, 8, 10, 12],
			twoHanded: false,
			armorPiercing: true
		}
	],
	inventory: [
		{
			name: "Hammer",
			notes: "",
			quantity: 1,
			slots: 1
		},
		{
			name: "Knife",
			notes: "",
			quantity: 1,
			slots: 1
		},
		{
			name: "Portable grill",
			notes: "",
			quantity: 1,
			slots: 1
		},
		{
			name: "Steaks",
			notes: "premium quality",
			quantity: 7,
			slots: 1
		},
		{
			name: "Flask of oil",
			notes: "",
			quantity: 1,
			slots: 1
		},
		{
			name: "Lantern",
			notes: "",
			quantity: 1,
			slots: 1
		}
	],
	wearing: "Waistcoat",
	money: 12,
	provisions: 6
};
