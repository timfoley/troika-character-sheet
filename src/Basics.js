import React from "react";

import "./Basics.css";

export function Basics(props) {
	return (
		<div className="basics">
			<label className="basics__name">
				Name
				<input type="text" value={props.name}/>
			</label>
			<label className="basics__background">
				Background
				<input type="text" value={props.background} />
			</label>
			<label className="basics__special">
				Special
				<textarea rows="4" value={props.special} />
			</label>
		</div>
	);
}
