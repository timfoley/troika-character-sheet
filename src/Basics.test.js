import "@testing-library/jest-dom/extend-expect";
import React from 'react';
import ReactDOM from 'react-dom';
import { getQueriesForElement } from "@testing-library/dom";

import char from "./char";
import { Basics } from "./Basics";

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Basics  character={char} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

it("renders a text input with a label 'Name'", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Basics  character={char} />, div);
    const { getByLabelText } = getQueriesForElement(div);
    const input = getByLabelText(/name/i);
    expect(input).toHaveAttribute("type", "text");
})