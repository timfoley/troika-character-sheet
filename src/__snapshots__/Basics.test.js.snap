// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="basics"
>
  <label
    class="basics__name"
  >
    Name
    <input
      class="basics__input basics__input--name"
      type="text"
      value=""
    />
  </label>
  <label
    class="basics__background"
  >
    Background
    <input
      class="basics__input basics__input--background "
      type="text"
      value=""
    />
  </label>
  <label
    class="basics__special"
  >
    Special
    <textarea
      class="basics__input basics__input--special"
      rows="4"
    />
  </label>
</div>
`;
